### ISSUE TITLE
Question - What is the best way to debug modules?

### ISSUE THREAD

**Issue by anwaramoon** [VirtoCommerce/vc-community#135]
I've been reading docs [here](http://docs.virtocommerce.com/display/vc2devguide/Creating+new+module) but I could not find a best way to debug a module. Are there any best practices? I know if I have written a custom module than I can write unit tests but if I had to debug existing modules than what is the recommended option.

Thanks

**Comment by Woland2k** [comment]
Compile solution in debug mode, then attach to IIS Worker Process from VS.NET and add breakpoints.

**Comment by anwaramoon** [comment]
Thanks for this info. I think I could not explain my question. Let me rephrase it.

I am working with solution VirtoCommerce.Manager.

My startup project is VirtoCommerce.Platform.Web.

When I start above project in debug mode and attach debugger to IIS. I can only debug code exists in VirtoCommerce.Platform.Web project or linked projects. But none of the commerce modules are being referenced in VirtoCommerce.Platform.Web project. All of the modules exists in App_Data\Modules folder. So what is the best way to debug commerce modules in development environment. For instance, if I want to debug content module (VirtoCommerce.Content.Web) than do I have to set startup project VirtoCommerce.Content.Web and attach debugger. This way it is very difficult to debug platform as you can only debug one module at a time.

Does it make sense or am I asking very stupid questions?

Thanks for the help.

**Comment by Woland2k** [comment]
First you need to configure solution correctly, meaning you should have a /modules virtual directory at the root project in IIS pointing to Modules directory in the solution just as described in http://docs.virtocommerce.com/display/vc2devguide/Source+Code+Getting+Started. 

Seocnd, you won't be able to start the project within VS.NET, you need to setup actual IIS (not IIS Express). You will then open the admin site in the browser. Something like "http://localhost/admin". The admin site should open and load all the modules.

You can then open VS.NET (in Administrator mode) and attach to w3wp.exe process. 
![image](https://cloud.githubusercontent.com/assets/1566470/11198540/4d91d288-8c7a-11e5-9ee1-0d7192ca8dce.png)

After that you should be able to add breakpoints and debug any module that exists in VS.NET solution.

**Comment by anwaramoon** [comment]
Bundle of thanks @Woland2k.

Everything was correct except Modules virtual directory path. By mistake it was pointing to App_Data folder instead of the Platform\Modules folder.

**Comment by Woland2k** [comment]
np, glad to help
